app/src/main/java/net/gsantner/markor/model/Document.java

```diff
@@ -324,13 +347,17 @@ public synchronized boolean saveContent(final Activity context, final CharSequen
                         if (isContentResolverProxyFile) {
                             GsFileUtils.writeFile(_file, contentAsBytes, _fileInfo);
                         }
-                    } catch (Exception ignored) {
+                    } catch (Exception e) {
+                        Log.i(Document.class.toString(), e.getMessage());
                     }
                 });
                 success = true;
             } else {
                 success = GsFileUtils.writeFile(_file, contentAsBytes, _fileInfo);
             }
+
+            success &= fileBytes() >= contentAsBytes.length;
```

Comment:
File size on disk must be greater than content size for success - another defensive check